app/file.go

```diff
@@ -397,6 +397,25 @@ func (a *App) UploadFiles(teamId string, channelId string, userId string, files
 	return resStruct, nil
 }
 
+// UploadFile uploads a single file in form of a completely constructed byte array for a channel.
+func (a *App) UploadFile(data []byte, channelId string, filename string) (*model.FileInfo, *model.AppError) {
+	info, _, appError := a.DoUploadFileExpectModification(time.Now(), "noteam", channelId, "nouser", filename, data)
```

Comment:
I think passing "noteam" makes sense, but I wonder if we should pass an optional user ID? Thoughts @chclaus @hanzei ?

Reply:
If the function is called from a plugin, I think you won't have an idea of a user. The fact is, that the normalized filepath will respect both, the team and the given user. Please take a look at app/file.go:448

```
pathPrefix := now.Format("20060102") + "/teams/" + teamId + "/channels/" + channelId + "/users/" + userId + "/" + info.Id + "/"
info.Path = pathPrefix + filename
```

In my eyes the consistence of the file directory structure is more important than the abstinence of a new synthetic user. Your thoughts @jwilander @hanzei ?